hack/tools/go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/metal3-io/cluster-api-provider-metal3/hack/tools
 
-go 1.23.0
+go 1.24.0
+
+toolchain go1.24.4
```

Comment:
I think we were not supposed to add toolchain directives? lets us go 1.24.4 instead if it is dictated by dependencies?